airflow/secrets/base_secrets.py

```diff
@@ -21,8 +21,11 @@
 from typing import TYPE_CHECKING
 
 from airflow.exceptions import RemovedInAirflow3Warning
+from airflow.utils.session import provide_session
 
 if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
```

Comment:
I think this is the problem. I am not sure removing type checking will solve the problem or cause more. If there won't be a session parameter at the base class that some implementations have, we can not pass the session from Variable and assume that there will always be a session parameter . Otherwise, it would fail.

Reply:
Removed